js/widget-form.js

```diff
@@ -10,9 +10,125 @@ if ( ! wp.widgets.formConstructor ) {
+			widgetForm._notificationsTemplate = wp.template( widgetForm.config.notifications_template_id );
+		}
+		return widgetForm._notificationsTemplate;
+	}
+
+	function renderNotificationsTemplate( container, templateFunction, notifications, altNotice ) {
+		container.empty().append( $.trim(
+			templateFunction( { notifications: notifications, altNotice: Boolean( altNotice ) } )
+		) );
+	}
+
+	function removeSanitizeNotifications( notifications ) {
+		notifications.each( function iterateNotifications( notification ) {
+			if ( notification.viaWidgetFormSanitizeReturn ) {
+				notifications.remove( notification.code );
+			}
+		} );
+	}
+
+	function addSanitizeNotification( widgetForm, notification ) {
+		notification.viaWidgetFormSanitizeReturn = true;
+		widgetForm.notifications.add( notification.code, notification );
+	}
+
+	function getValidateWidget( widgetForm, previousValidate ) {
+		/**
+		 * Validate the instance data.
+		 *
+		 * @todo In order for returning an error/notification to work properly, api._handleSettingValidities needs to only remove notification errors that are no longer valid which are fromServer:
+		 *
+		 * @param {object} value Instance value.
+		 * @returns {object|Error|wp.customize.Notification} Sanitized instance value or error/notification.
+		 */
+		return function validate( value ) {
+			var setting = this, newValue, oldValue;
+			oldValue = setting();
+			newValue = widgetForm.sanitize( previousValidate.call( setting, value ), oldValue );
+
+			// Remove all existing notifications added via sanitization since only one can be returned.
+			removeSanitizeNotifications( widgetForm.notifications );
+
+			// If sanitize method returned an error/notification, block setting update and add a notification
+			if ( newValue instanceof Error ) {
+				newValue = new api.Notification( 'invalidValue', { message: newValue.message, type: 'error' } );
+			}
+			if ( newValue instanceof api.Notification ) {
+				addSanitizeNotification( widgetForm, newValue );
+				return null;
+			}
+
+			return newValue;
+		};
+	}
+
+	function validateForm( widgetForm ) {
+		if ( ! widgetForm.model || ! widgetForm.model.extended || ! widgetForm.model.extended( api.Value ) ) {
+			throw new Error( 'Widget Form is missing model property which must be a Value or Setting instance.' );
+		}
+		if ( 0 === widgetForm.container.length ) {
+			throw new Error( 'Widget Form is missing container property as Element or jQuery.' );
+		}
+	}
+
+	function getValidatedFormProperties( config, properties ) {
+		var defaultConfig = {
+			form_template_id: '',
+			notifications_template_id: '',
+			l10n: {},
+			default_instance: {}
+		};
+
+		var defaultProperties = {
+			model: null,
+			container: null,
+			config: {}
+		};
+
+		var args = _.extend( {}, defaultProperties, properties || {} );
+		var propertiesConfig = properties ? properties.config : {};
+		args.config = _.extend( {}, defaultConfig, config, propertiesConfig || {} );
+		return args;
+	}
```

Comment:
Added in bafd1b9. Also, I realize these private functions kind of clutter up the top of the file. I just put them there for convenience. They could just as easily be in their own "namespace" object and/or at the bottom of the file and/or in a separate file. I'm very open to other thoughts about that.

They could even be function properties on the `Form` object, but I'd suggest against that unless there's a reason they would be called directly or overridden (making them private is currently no different than having their functionality inside of the current public functions).